### PR TITLE
Feature/onlineatps

### PR DESCRIPTION
**Pull request by jonaprieto**
This gives support to use [onlineatps](https://github.com/jonaprieto/onlineatps).

### CODE REVIEW

**Comment by jonaprieto** [comment]
I don't how to fix this error in travis. It's not clear. Can you help me?

**Comment by asr** [comment]
> I don't how to fix this error in travis. It's not clear. Can you help me?

Could you reproduce locally the error? That is, which is the output of `make non_conjectures`?

**Comment by asr** [comment]
I don't understand why your PR includes some commits from the master branch. Create a "clean" PR please.

File: README.md [hunk 1]
```diff
@@ -110,6 +114,20 @@ Command-line options
    Equinox, version 5.0alpha, 2010-06-29 proved the conjecture
    ````
 
+  If we want to use a certain online ATP, say E, we must install OnlineATPs.
```
Comment (asr) [inline README.md]: Add a link to  `OnlineATPs` please.

**Comment by asr** [comment]
> I don't understand why your PR includes some commits from the master branch. Create a "clean" PR please.

Forget what I said.

**Comment by asr** [comment]
Using the `Test.agda` file in the README.md, I got the following results:

```
$ apia --atp=e Test.agda
Proving the conjecture in /tmp/Test/9-8744-comm.fof
E 1.9 Sourenee proved the conjecture
```

```
apia --atp=online-e Test.agda
Proving the conjecture in /tmp/Test/9-8744-comm.fof
E---2.0 *did not* prove the conjecture
apia: the ATP(s) did not prove the conjecture in /tmp/Test/9-8744-comm.fof
```

That is, I couldn't prove the conjecture using the `--atp=online-e` option.

File: test/succeed/non-conjectures/non-conjectures.test [hunk 2]
```diff
@@ -106,7 +107,7 @@ Interface
   current = Foo
   context = TopCtx
   modules
-    scope 
+    scope
```
Comment (asr) [inline test/succeed/non-conjectures/non-conjectures.test]: The `make non_conjectures` command is failing due to this change. Revert it please.

File: README.md [hunk 3]
```diff
@@ -110,6 +114,24 @@ Command-line options
...
+  [SystemOnTPTP](http://www.cs.miami.edu/~tptp/cgi-bin/SystemOnTPTP).
+  Therefore, we could use a version of E Prover  with `onlineatps` installed
+  using the following command:
+
+  ````bash
+  $ apia --atp=online-e Test.agda
+  Proving the conjecture in /tmp/Test/9-8744-comm.tptp ...
+  E-2.0 proved the conjecture
+  ````
+  To see a list of all online atps available, run the following command:
```
Comment (asr) [inline README.md]: Typo atps -> ATPs

File: src/Apia/Common.hs [hunk 4]
```diff
@@ -20,13 +20,16 @@ data ATP = CVC4
...
+         | OnlineATP String
          | SPASS
          | Vampire
          | Z3
          deriving Show
 
 instance Pretty ATP where
-  pretty = text . show
+  pretty (OnlineATP atp) = text . show $ atp
+  pretty other = text . show $ other
```
Comment (asr) [inline src/Apia/Common.hs]: Fix the indentation.

**Comment by asr** [comment]
Good job! Please fix the small details I pointed-out and I'll merge the PR.

**Comment by asr** [comment]
The PR has conflicts with master, so I cannot merge it. Please push a commit fixing the conflicts.

File: README.md [hunk 5]
```diff
@@ -110,6 +114,24 @@ Command-line options
...
+  [SystemOnTPTP](http://www.cs.miami.edu/~tptp/cgi-bin/SystemOnTPTP).
+  Therefore, we could use a version of E Prover  with `onlineatps` installed
+  using the following command:
+
+  ````bash
+  $ apia --atp=online-e Test.agda
+  Proving the conjecture in /tmp/Test/9-8744-comm.tptp ...
+  E-2.0 proved the conjecture
+  ````
+  To see a list of all online ATPS available, run the following command:
```
Comment (asr) [inline README.md:129]: Typo: ATPS -> ATPs

**Comment by jonaprieto** [comment]
I will open a new PR. It is not easy to solve the conflicts.

**Pull request closed without merging**